modin/pandas/dataframe.py

```diff
@@ -2544,6 +2545,23 @@ def __setitem__(self, key, value):
                     key = DataFrame(key, columns=self.columns)
                 return self.mask(key, value, inplace=True)
 
+            if isinstance(key, list) and all((x in self.columns for x in key)):
```

Comment:
`write_items` function does not work with labels that are not in the index.

Reply:
`loc` is not quite suitable, reindexing works differently, discussion in another thread (https://github.com/modin-project/modin/pull/4455/files#r914827282).